crates/duty-tracker/src/contract_manager.rs

```diff
@@ -210,6 +234,12 @@ struct ContractManagerCtx {
     stake_chain_persister: StakeChainPersister,
     active_contracts: BTreeMap<Txid, ContractSM>,
     stake_chains: StakeChainSM,
+    wallet: OperatorWallet,
+    s2_client: SecretServiceClient,
+    /// NOTE: DO NOT CALL .next() because it will mess with the contract manager and break shit
+    p2p_msg_handle: MessageHandler,
```

Comment:
VERY VERY important.
We might need to refactor this in the future.
Dunno, maybe different types, one that has an exclusive (mutable) ref or ownership of the `P2PHandle` for the ContractManager and another that has shared (uncuttable) ref of the `P2PHandle` so that it cannot call `.next` by any means.